master/buildbot/reporters/authz/irc.py

```diff
@@ -0,0 +1,33 @@
+from buildbot.reporters.authz import Authz
+
+
+class IrcAuthz(Authz):
+
+    def __init__(self, allowOnlyOps=True):
+        self.channelUsers = {}
+        self.allowOnlyOps = allowOnlyOps
+
+    def addUserstoChannel(self, nicklist, channel):
+        if '%s-tmp' % channel not in self.channelUsers:
+            self.channelUsers['%s-tmp' % channel] = []
+        n = self.channelUsers['%s-tmp' % channel]
+        n += nicklist
+
+    def finalizeAddUsersToChannel(self, channel):
+        if '%s-tmp' % channel not in self.channelUsers:
+            return
+
+        self.channelUsers[channel] = self.channelUsers['%s-tmp' % channel]
+        del self.channelUsers['%s-tmp' % channel]
+
+    def assertAllowPM(self):
+        if self.allowOnlyOps:
+            return False
```

Comment:
This is possiblle. It means we'll have to go through the list of cannels (on this network) for the ops status of a user.